src/subsidy_base.h

```diff
@@ -52,11 +52,18 @@ struct Subsidy : SubsidyPool::PoolItem<&_subsidy_pool> {
 
 /** Constants related to subsidies */
 static const uint SUBSIDY_OFFER_MONTHS         =  12; ///< Duration of subsidy offer
-static const uint SUBSIDY_CONTRACT_MONTHS      =  12; ///< Duration of subsidy after awarding
 static const uint SUBSIDY_PAX_MIN_POPULATION   = 400; ///< Min. population of towns for subsidised pax route
 static const uint SUBSIDY_CARGO_MIN_POPULATION = 900; ///< Min. population of destination town for cargo route
 static const uint SUBSIDY_MAX_PCT_TRANSPORTED  =  42; ///< Subsidy will be created only for towns/industries with less % transported
 static const uint SUBSIDY_MAX_DISTANCE         =  70; ///< Max. length of subsidised route (DistanceManhattan)
 static const uint SUBSIDY_TOWN_CARGO_RADIUS    =   6; ///< Extent of a tile area around town center when scanning for town cargo acceptance and production (6 ~= min catchmement + min station / 2)
 
+/** Types of subsidy news messages, which determine how the date is printed and whether to use singular or plural cargo names */
+enum SubsidyDecodeParamType {
+	SUBSIDY_NEWS_OFFERED   = 0, ///< News item for an offered subsidy
+	SUBSIDY_NEWS_AWARDED   = 1, ///< News item for an awarded subsidy
+	SUBSIDY_NEWS_WITHDRAWN = 2, ///< News item for a subsidy offer withdrawn, or expired subsidy
+	SUBSIDY_GUI            = 3, ///< Subsidies listed in the Subsidy GUI
```

Comment:
I think this name is confusing. Maybe prefix it with `DPARAM_`, so `DPARAM_SUBSIDY_GUI`? Dunno .. but these names are asking for trouble :)

Alternatively, use class enums (https://en.cppreference.com/w/cpp/language/enum under `Scoped enumerations`).